app/controllers/users_controller.rb

```diff
@@ -34,6 +36,20 @@ def destroy
   def index
   end
   
+  # Confirms a logged-in user.
+  def logged_in_user
+    unless logged_in?
+      flash[:danger] = "Please log in."
+      redirect_to login_path
+    end
+  end
+  
+  # Confirms the correct user.
+  def correct_user
+    @user = User.find(params[:id])
```

Comment:
This feels redundant. `session[:user_id]` can only hold one value and thus only map to one user. So you're either logged in or you're not, right? @rynkwn